### PR TITLE
Add FP16 converter helper, and then support UInt8FusedQTy to UInt8FusedFP16QTy conversion with it.

### PR DESCRIPTION
**Pull request by jfix71**
Summary: We want to support converting UInt8FusedQTy to UInt8FusedFP16QTy when we do FP16 conversion. This does not fit naturally into the TypeAToTypeBFunctionConverter. Instead of making major modifications/redesigning it, I added a helper to wrap the TypeAToTypeBFunctionConverter with Float -> Float16 with a secondary pass to convert UInt8FusedQTy to UInt8FusedFP16QTy.

I additionally added support for the graph optimization for ConvertTo to merge a Constant UInt8FusedQTy followed by a ConvertTo UInt8FusedFP16QTy into the Constant.

Test Plan: Added a test for the new graph optimization case that is supported. Added an OperatorTest and a conversion test to test that converting a FRWQ-SLWS to FP16 works correctly.

### CODE REVIEW

File: include/glow/Base/Tensor.h [hunk 1]
```diff
@@ -573,6 +573,9 @@ class Tensor final {
   /// Convert each element of this tensor to \p newTy.
   void convertToType(ElemKind newTy);
```
Comment (rdzhabarov) [inline include/glow/Base/Tensor.h]: could you update the comment on what's currently supported, in terms of typea -> typeb conversion

File: lib/Base/Tensor.cpp [hunk 2]
```diff
@@ -543,20 +543,68 @@ void Tensor::init(InitKind init, float val, PseudoRNG &PRNG) {
...
-    llvm_unreachable("Type not supported");
+  *this = this->getCopyConvertedToType(newTy);
+}
+
+Tensor Tensor::getCopyConvertedToType(ElemKind newKind) const {
+  if (!isQuantizedElemKind(newKind)) {
+    Tensor tmp(newKind, dims());
+    switch (newKind) {
+    case ElemKind::Float16Ty:
+      assert(getElementType() == ElemKind::FloatTy && "Cast not implemented");
```
Comment (rdzhabarov) [inline lib/Base/Tensor.cpp]: could we switch to DCHECK and add what exact cast is not implemented. Like Cast from X to Y is not implemented.

File: lib/Base/Tensor.cpp [hunk 3]
```diff
@@ -543,20 +543,68 @@ void Tensor::init(InitKind init, float val, PseudoRNG &PRNG) {
...
+    return tmp;
+  }
+
+  // Handle Fused conversion. Currently only supports UInt8FusedQTy ->
+  // UInt8FusedFP16QTy.
+  assert((getElementType() == ElemKind::UInt8FusedQTy) &&
+         (newKind == ElemKind::UInt8FusedFP16QTy) &&
+         "Conversion not implemented.");
+  Tensor tmp(newKind,
+             {dims()[0], dims()[1] - 2 * (sizeof(float) - sizeof(float16_t))},
```
Comment (rdzhabarov) [inline lib/Base/Tensor.cpp]: can we make an assumption here that dims().size() == 2?

File: lib/Graph/Nodes.cpp [hunk 4]
```diff
@@ -553,10 +553,38 @@ bool Convolution3DGradNode::verify() const {
...
+
 bool ConvertToNode::verify() const {
-  bool isValid = checkSameShape(getInput(), getResult(), this);
   TypeRef srcTy = getInput().getType();
   TypeRef dstTy = getResult().getType();
+  const bool srcIsFused = isFusedQuantizedElemKind(srcTy->getElementType());
+  const bool dstIsFused = isFusedQuantizedElemKind(dstTy->getElementType());
+
+  bool isValid = expectCompareTrue(
+      "Must convert between Fused ElemKinds.",
```
Comment (rdzhabarov) [inline lib/Graph/Nodes.cpp]: a little bit incomplete, since it's ok for both to be unfused.

File: tests/unittests/BackendTestUtils.cpp [hunk 5]
```diff
@@ -1131,4 +1131,81 @@ void runOnDevice(ExecutionContext &context, llvm::StringRef name,
...
+static void copyInScaleOffset(char *currRowScaleOffsetPtr) {
+  // Range (0, 255) -> (-0.1, 0.1)
+  T scale = 1.0f / 1275;
+  T offset = -0.1;
+
+  memcpy(currRowScaleOffsetPtr, &scale, sizeof(T));
+  memcpy(currRowScaleOffsetPtr + sizeof(T), &offset, sizeof(T));
+}
+
+Constant *createRandomizedConstant(Module &mod, TypeRef type,
```
Comment (rdzhabarov) [inline tests/unittests/BackendTestUtils.cpp]: wanted to put a comment to reuse this in recsys and noticed it's down below :) nice.

File: lib/Base/Tensor.cpp (hunk 3 above)
Comment (jfix71) [reply lib/Base/Tensor.cpp]: There are asserts in other places checking this for the Fused ElemKinds, but I can add a new check for it here as well.

**Comment by jfix71** [comment]
@rdzhabarov All comments addressed!

**Pull request closed without merging**